### PR TITLE
Adds deep_merge Ruby implementation

### PR DESCRIPTION
**Pull request by adamof**
Solves https://github.com/paliarush/magento2-vagrant-for-developers/issues/124

Adds an implementation for the `deep_merge` method that comes from Ruby on Rails usually.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -4,6 +4,7 @@
 Vagrant.require_version ">= 1.8"
 
 require 'yaml'
+require './deep_merge'
```
Comment (paliarush) [inline Vagrantfile:7]: How about `require 'vagrant/util/deep_merge'` and then `Vagrant::Util::DeepMerge.deep_merge(config_data_dist, config_data)` ?

It seems simpler and the implementation of `deep_merge` will be supported by vagrant developers.

See https://github.com/hashicorp/vagrant/commit/4c9e6e46b2a6cf7b7949ab85e960957fdf5faf83

Comment (adamof) [reply Vagrantfile:7]: Even better! Will do

Comment (paliarush) [reply Vagrantfile:7]: @adamof will you be able to contribute the proposed fix in the nearest days? If you don't have time, I will work deliver it

Comment (adamof) [reply Vagrantfile:7]: Hey! Go ahead! I'm laptop-less at the moment :/

Comment (paliarush) [reply Vagrantfile:7]: Sounds good, thanks for contributing anyway!

**Pull request closed without merging**